Automattic/wp-calypso
# Signup: remove site information component and update site title

## Changes proposed in this Pull Request

### Using site title component in the flow

Since we're no longer capturing `phone` and `address` values, we don't need the site information component.

Instead, let's update the site title component with the latest styles, and plug its value into the state.

### Removing the site information component
We're also removing the site information component and all references to it (steps and flow refs as well)  in favour of the site title component.

### Retiring the site information Tracks event

We'll use `calypso_signup_actions_submit_site_title` in the place of `calypso_signup_actions_submit_site_information`.

👉 @megsfulton @niranjan-uma-shankar @fditrapani @adamziel You're using `calypso_signup_actions_submit_site_information` in some recent funnels I see. If this gets merged, that event will no longer fire. Let me know if there are any issues. Thanks!

## Testing instructions

Create a site and fill out the site title step. Check the following:

1. That we're saving `siteTitle` to the `state.signup.steps.siteTitle`
2. When you submit the site title step, we're firing `calypso_signup_actions_submit_site_title` tracking event with `{ value: 'your site title' }`
3. After creating a site, the site title you entered appears on your site.

Since #32432 we're not using the `site-information-without-domains` step, but just to make sure, test the flow using the `remove` variation of the `removeDomainsStepFromOnboarding` AB test –

<img width="600" alt="Screen Shot 2019-04-25 at 2 36 39 pm" src="https://user-images.githubusercontent.com/6458278/56710210-57465e80-6768-11e9-9ba3-5543c250cf90.png">

– and repeat the steps above. 👍 We've created site-title-without-domain to replace it.
